.github/workflows/rust.yml

```diff
@@ -90,28 +87,14 @@ jobs:
         TARGET: x86_64-unknown-linux-gnu
       run: sh ci/run.sh
 
-  # These jobs doesn't actually test anything, but they're only used to tell
-  # bors the build completed, as there is no practical way to detect when a
-  # workflow is successful listening to webhooks only.
-  #
-  # ALL THE PREVIOUS JOBS NEED TO BE ADDED TO THE `needs` SECTION OF THIS JOB!
-
-  end-success:
-    name: bors build finished
-    if: github.event.pusher.name == 'bors' && success()
-    runs-on: ubuntu-latest
-    needs: [miri, rustfmt_clippy, test, msrv]
-
-    steps:
-      - name: Mark the job as successful
-        run: exit 0
-
-  end-failure:
-    name: bors build finished
-    if: github.event.pusher.name == 'bors' && (failure() || cancelled())
+  conclusion:
+    needs: [test, msrv]
+    # !cancelled() executes the job regardless of whether the previous jobs passed, failed or get skipped.
+    if: ${{ !cancelled() }}
     runs-on: ubuntu-latest
-    needs: [miri, rustfmt_clippy, test, msrv]
-
     steps:
-      - name: Mark the job as a failure
-        run: exit 1
+      - name: Conclusion
+        run: |
+          jq -C <<< '${{ toJson(needs) }}'
+          # Check if all needs were successful.
+          jq --exit-status 'all(.result == "success")' <<< '${{ toJson(needs) }}'
```

Comment:
copy pasted the same blob we added in other repos
https://github.com/rust-lang/rust-clippy/blob/8bc1a9e49a22b0e43c942102b13ded8704851228/.github/workflows/clippy_pr.yml#L62